spec/tbd_tests_spec.rb

```diff
@@ -11543,4 +11549,570 @@
+      expect(heat.is_a?(Hash)).to be( true)
+      expect(cool.is_a?(Hash)).to be( true)
+      expect(heat.key?(:spt )).to be( true)
+      expect(cool.key?(:spt )).to be( true)
+      expect(heat.key?(:dual)).to be( true)
+      expect(cool.key?(:dual)).to be( true)
+      expect(heat[:spt ].nil?).to be( true)
+      expect(cool[:spt ].nil?).to be( true)
+      expect(heat[:dual]     ).to be(false)
+      expect(cool[:dual]     ).to be(false)
+
+      expect(TBD.plenum?(attic, loops, setpoints)).to be(false)
+
+      # Add a dual setpoint temperature schedule.
+      identifier = "TEMPORARY attic setpoint schedule"
+      sched = OpenStudio::Model::ScheduleCompact.new(model)
+      sched.setName(identifier)
+      expect(sched.constantValue.empty?                        ).to be(true)
+      expect(tstat.setHeatingSetpointTemperatureSchedule(sched)).to be(true)
+
+      # After the addition.
+      expect(tstat.getHeatingSchedule.empty?).to be(false)
+      expect(tstat.getCoolingSchedule.empty?).to be( true)
+      heat = TBD.maxHeatScheduledSetpoint(zone)
+
+      expect(heat.nil?       ).to be(false)
+      expect(heat.is_a?(Hash)).to be( true)
+      expect(heat.key?(:spt )).to be( true)
+      expect(heat.key?(:dual)).to be( true)
+      expect(heat[:spt ].nil?).to be( true)
+      expect(heat[:dual]     ).to be( true)
+
+      expect(TBD.plenum?(attic, loops, setpoints)).to be(true) # works ...
+
+      json      = TBD.process(model, argh)
+      expect(json.is_a?(Hash)    ).to be( true)
+      expect(json.key?(:io      )).to be( true)
+      expect(json.key?(:surfaces)).to be( true)
+      io        = json[:io      ]
+      surfaces  = json[:surfaces]
+      expect(TBD.error?          ).to be( true)
+      expect(TBD.logs.empty?     ).to be(false)
+
+      # The incomplete (temporary) schedule triggers a non-FATAL TBD error.
+      TBD.logs.each do |log|
+        expect(log[:message].include?("Empty '"                 )).to be(true)
+        expect(log[:message].include?("::scheduleCompactMinMax)")).to be(true)
+      end
+
+      expect(surfaces.nil?       ).to be(false)
+      expect(surfaces.is_a?(Hash)).to be( true)
+      expect(surfaces.size       ).to eq(   18)
+      expect(io.nil?             ).to be(false)
+      expect(io.is_a?(Hash)      ).to be( true)
+      expect(io.empty?           ).to be(false)
+      expect(io.key?(:edges     )).to be( true)
+
+      surfaces.values.each do |surface|
+        expect(surface.is_a?(Hash)        ).to be(true)
+        expect(surface.key?(:conditioned )).to be(true)
+        expect(surface.key?(:deratable   )).to be(true)
+        expect(surface.key?(:construction)).to be(true)
+        expect(surface.key?(:ground      )).to be(true)
+
+        next     if surface[:ground   ]
+        next unless surface[:deratable]
+
+        id = surface[:construction].nameString
+        ok = id.include?("BTAP-Ext-Roof") || id.include?("BTAP-Ext-Wall")
+        expect(ok).to be(true)
```

Comment:
Without the temporary (albeit incomplete) schedule, TBD would have derated the (here, uninsulated) attic floor. Now, only the exterior roof is derated.